x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationListItem.java

```diff
@@ -75,6 +81,30 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    private static final ConstructingObjectParser<SearchApplicationListItem, String> PARSER = new ConstructingObjectParser<>(
+        "search_application_list_item`",
+        false,
+        (params) -> {
+            final String name = (String) params[0];
+            @SuppressWarnings("unchecked")
+            final String[] indices = ((List<String>) params[1]).toArray(String[]::new);
+            final String analyticsCollectionName = (String) params[2];
+            final Long updatedAtMillis = (Long) params[3];
+            return new SearchApplicationListItem(name, indices, analyticsCollectionName, updatedAtMillis);
+        }
+    );
+
+    static {
+        PARSER.declareStringOrNull(optionalConstructorArg(), NAME_FIELD);
+        PARSER.declareStringArray(constructorArg(), INDICES_FIELD);
```

Comment:
just a note for both of us - this will not be compatible with #98036 since I removed the `indices` from the list search apps API - so whichever PR gets merged second, we need to make sure we get the latest changes

Reply:
That's a really good observation. I actually think it might be best to merge this one first, so we can ensure that BWC works after removing the indices. WDYT?